lib/matplotlib/__init__.py

```diff
@@ -619,9 +620,30 @@ def get_cachedir():
     return _get_config_or_cache_dir(_get_xdg_cache_dir())
 
 
-def _get_data_path():
-    """Return the path to matplotlib data."""
+@_logged_cached('matplotlib data path: %s')
+def get_data_path(*, _from_rc=None):
+    """Return the path to Matplotlib data."""
+    if _from_rc is not None:
+        cbook.warn_deprecated(
+            "3.2",
+            message=("Setting the datapath via matplotlibrc is "
+            "deprecated %(since)s and will be removed in %(removal)s. "
+            ""),
+            removal='3.3')
+        path = Path(_from_rc)
+        if path.is_dir():
+            return str(path)
+        else:
+            warnings.warn(f"You passed datapath: {path!r} in your "
```

Comment:
Do you really want `!r` here, as `path` is a `PosixPath`/`WindowsPath`? Maybe using `{_from_rc}` would be better

Reply:
Thought was to let it be shown with any cleanup, but I also see the case for showing the version that is in their files (as it is then easier to grep for).